channels/api.py

```diff
@@ -683,7 +687,76 @@ def update_channel(self, name, title=None, channel_type=None, **other_settings):
         return self.get_channel(name)
 
     @staticmethod
-    def _apply_vote(instance, validated_data, allow_downvote=False, instance_type=None):
+    def _apply_comment_vote(comment, upvote, downvote, is_upvoted, is_downvoted):
+        """
+        Applies an upvote to the related Comment record
+
+        Args:
+            comment (Comment): the reddit comment
+            upvote (boolean): nullable flag for applying an upvote
+            downvote (boolean): nullable flag for applying a downvote
+            is_upvoted (boolean): True if the user currently has upvoted the post
+            is_downvoted (boolean): True if the user currently has downvoted the post
+        """
+        vote_delta = sum(
+            [
+                # start handle upvote
+                # adds the upvote
+                1 if upvote and not is_upvoted else 0,
+                # removes an opposing downvote, if present
+                1 if upvote and is_downvoted else 0,
+                # clear an existing upvote only, but only count this once if downvoting also
+                -1 if upvote is False and is_upvoted and not downvote else 0,
+                # end handle upvote
+                # --------
+                # start handle downvote
+                # adds the downvote
+                -1 if downvote and not is_downvoted else 0,
+                # removes an opposing upvote, if present
+                -1 if downvote and is_upvoted else 0,
+                # clear an existing downvote, but only count this once if upvoting also
+                1 if downvote is False and is_downvoted and not upvote else 0,
+                # end handle downvote
```

Comment:
Reason for this is because `vote_action` erases a critical piece of information we need here: "does an upvote need to increment by 1 or is there a current downvote and we need to increment by 2?"